sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/SerializerEncoding.java

```diff
@@ -20,21 +25,110 @@
+            return JSON;
+        }
+
+        final String[] parts = mimeContentType.split(";");
+
+        final String mimeType = parts[0].toLowerCase(Locale.ROOT);
+        final String[] mimeTypeParts = mimeType.split("/");
+        if (mimeTypeParts.length != 2) {
+            LOGGER.warning("Content-Type '{}' does not match mime-type formatting 'type/'subtype'. "
+                + "Returning default: {}", mimeType, DEFAULT_ENCODING);
+            return DEFAULT_ENCODING;
+        }
+
+        for (SerializerEncodingMimeTypes serializerEncodingMimeTypes : SUPPORTED_MIME_TYPES) {
+            if (serializerEncodingMimeTypes.matches(mimeTypeParts[0], mimeTypeParts[1])) {
+                return serializerEncodingMimeTypes.getEncoding();
+            }
+        }
+
+        LOGGER.warning("Content-Type '{}' does not match any supported one. Returning default: {}",
+            mimeType, DEFAULT_ENCODING);
+
+        return DEFAULT_ENCODING;
+    }
+
+    /**
+     * A mapping of serialization type to the supported mime type.
+     */
+    private static final class SerializerEncodingMimeTypes {
+        private final SerializerEncoding encoding;
+        private final String suffix;
+        private final List<MimeType> mimeTypes;
+
+        private SerializerEncodingMimeTypes(SerializerEncoding encoding, String suffix, MimeType... mimeTypes) {
+            this.encoding = encoding;
+            this.suffix = suffix;
+            this.mimeTypes = Arrays.asList(mimeTypes);
+        }
+
+        private boolean matches(String type, String subtype) {
+            for (MimeType supported : mimeTypes) {
+                if (supported.matches(type, subtype)) {
+                    return true;
+                }
             }
+
+            // Check the suffix if it does not match the full types.
+            final int lastIndex = subtype.lastIndexOf("+");
+            if (lastIndex == -1) {
+                return false;
+            }
+
+            final String mimeTypeSuffix = subtype.substring(lastIndex + 1);
+            return suffix.equals(mimeTypeSuffix);
         }
 
-        return JSON;
+        public SerializerEncoding getEncoding() {
+            return encoding;
+        }
+    }
+
+    private static final class MimeType {
+        private final String type;
+        private final String subType;
+
+        private MimeType(String rawMimeType) {
+            final String[] parts = rawMimeType.split("/");
+            if (parts.length != 2) {
+                throw new IllegalArgumentException("mimeType is not properly formatted with {type}/{subtype}: "
+                    + rawMimeType);
+            }
+
+            this.type = parts[0];
+            this.subType = parts[1];
+        }
+
+        private boolean matches(String otherType, String otherSubType) {
+            return type.equals(otherType) && subType.equals(otherSubType);
```

Comment:
I think MIME types are case insensitive but good to double-check.

Reply:
in line 59, I do a .toLowerCase(Locale.ROOT), but I can update this to be case insensitive instead.

Reply:
ah, I missed that, thanks for the reference.